packages/catalog-process/src/utilities/versionGenerator.ts

```diff
@@ -1,20 +1,33 @@
-import { EService } from "pagopa-interop-models";
+import { Descriptor, EService } from "pagopa-interop-models";
 import { z } from "zod";
 import { invalidDescriptorVersion } from "../model/domain/errors.js";
 
-export const nextDescriptorVersion = (eservice: EService): string => {
-  const currentVersion = eservice.descriptors.reduce((max, descriptor) => {
-    const currentVersionNumber = z.coerce
-      .number()
-      .safeParse(descriptor.version);
-    if (!currentVersionNumber.success) {
-      throw invalidDescriptorVersion(
-        `${descriptor.version} is not a valid descriptor version`
-      );
-    }
+function parseVersionNumber(version: string): number {
+  const versionNumber = z.coerce.number().safeParse(version);
+  if (!versionNumber.success) {
+    throw invalidDescriptorVersion(
+      `${version} is not a valid descriptor version`
+    );
+  }
+  return versionNumber.data;
+}
 
-    return currentVersionNumber.data > max ? currentVersionNumber.data : max;
-  }, 0);
+export const getLatestDescriptor = (
+  eservice: EService
+): Descriptor | undefined =>
+  eservice.descriptors.length > 0
+    ? eservice.descriptors.reduce(
+        (latest, descriptor) =>
+          parseVersionNumber(descriptor.version) >
+          parseVersionNumber(latest.version)
+            ? descriptor
+            : latest,
+        eservice.descriptors[0]
+      )
+    : undefined;
```

Comment:
Usually, in other parts of the code, we do as follows (or, if you prefer, let's use `parseVersionNumber` instead of `Number` so we can throw a meaningful error)

```suggestion
eservice.descriptors
    .sort((a, b) => Number(a.version) - Number(b.version))
    .at(-1);
```